File: app/channels/lesson_channel.rb
class LessonChannel < ApplicationCable::Channel
  def subscribed
    # transmit(params)
    stream_from "lesson_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    if(data["type"] == "message")
      data["from"] = data["from"].titleize
      # message_str = data["from"]+":"+data["message"]
      lesson = Lesson.find_by slug:data["lesson_slug"]
      lesson.messages << data
      lesson.save
      # puts data
      transmit(data)
    elsif(data["type"] == "slide")

    end
    # ActionCable.server.broadcast("InstructionChannel", data)
  end
end

File: app/javascript/channels/lesson_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("LessonChannel", {
  connected() {
    console.log("Lesson Channel connected.")
    if(!application.channels)
      application.channels = []

    application.channels.push(this)
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },
  received(data) {
    if(data["lesson_slug"] == $('meta[name=lesson-slug]').attr('id'))
    {
      application.receiveCallback(data)
    }
  }
});
